### DISCUSSION TITLE
Remove the 255 length limit on user attributes

### DISCUSSION THREAD

**Discussion by zhengyang1988** [keycloak/keycloak#20993] [Q&A]
We are using keycloak 18.0.2 in our production environment. We need to store something that is much longer than 255 characters as 
one user attribute on keycloak for each user.  But the length of  the value of a user attribute on keycloak can not exceed 255 characters(at least in the version 18.0.2). I found the following solution for this problem on stackoverflow:

https://stackoverflow.com/questions/44851052/max-size-of-custom-user-attribute-in-keycloak#:~:text=Keycloak%27s%20attribute%20values%20maximum%20length,the%20field%20type%20to%20text%20.

Is it good to solve the problem this way? Since this requires modify the data type of a database field, will it cause any trouble in the future? (I've already tested that this change won't cause any problem for the automatic db migration from 18.0.2 to 21.1.1 by runing keycloak 21.1.1 directly on the database after making this db column datatype change)

**Comment by jonkoops** [comment]
I believe user attributes are exposed on the token when authenticating, which is probably why there is a limit on the size. But I could be mistaken.

**Reply by sschu** [reply]
@jonkoops They could be exposed if you add an Attribute Mapper but normally they are not.
@zhengyang1988 You can just increase the column length, it is unlikely this causes problems (most likely situation would be when Keycloak itself changes the column type/length).
